dali/kernels/common/join/tensor_join_cpu_test.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include "dali/kernels/common/join/tensor_join_cpu.h"
+
+namespace dali {
+namespace kernels {
+namespace test {
+
+TEST(TensorJoinCpuTest, DetermineShapeStack) {
+  std::vector<TensorShape<>> shin = {{4, 5, 7, 8},
+                                     {4, 5, 7, 8}};
+  TensorShape<> sh0 = {2, 4, 5, 7, 8};
+  TensorShape<> sh1 = {4, 2, 5, 7, 8};
+  TensorShape<> sh2 = {4, 5, 2, 7, 8};
+  TensorShape<> sh3 = {4, 5, 7, 2, 8};
+  TensorShape<> sh4 = {4, 5, 7, 8, 2};
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 0), sh0);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 1), sh1);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 2), sh2);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 3), sh3);
+  EXPECT_EQ(impl::DetermineShape<true>(make_span(shin), 4), sh4);
+}
+
+
+TEST(TensorJoinCpuTest, DetermineShapeConcatConstantExtent) {
+  std::vector<TensorShape<>> shin = {{4, 5, 7, 8},
+                                     {4, 5, 7, 8}};
+  TensorShape<> sh0 = {8, 5, 7, 8};
+  TensorShape<> sh1 = {4, 10, 7, 8};
+  TensorShape<> sh2 = {4, 5, 14, 8};
+  TensorShape<> sh3 = {4, 5, 7, 16};
+  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 0), sh0);
+  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 1), sh1);
+  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 2), sh2);
+  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 3), sh3);
+}
+
+
+TEST(TensorJoinCpuTest, DetermineShapeConcatVaryingExtent) {
+  std::vector<TensorShape<>> shin = {{4, 5, 7, 8},
+                                     {4, 5, 3, 8}};
+  TensorShape<> shout = {4, 5, 10, 8};
+  EXPECT_EQ(impl::DetermineShape<false>(make_span(shin), 2), shout);
+}
+
+
+TEST(TensorJoinCpuTest, TransferBufferTest) {
+  using namespace std;  // NOLINT
+
+  vector<vector<int>> arr = {{6, 8, 5, 1, 3, 5, 1, 6, 8, 3, 7, 5},
+                             {4, 5, 1, 8, 4, 4, 1, 4, 1, 7, 6, 6}};
```

Comment:
Add some line breaks for readability:
```suggestion
  vector<vector<int>> arr = {{
    6, 8, 5, 1,
    3, 5, 1, 6,
    8, 3, 7, 5
  }, {
    4, 5, 1, 8,
    4, 4, 1, 4,
    1, 7, 6, 6
  }};
```

Reply:
Also: can you put unique numbers there? This way if anything goes wrong, we'll know.

Reply:
vector<vector<int>> arr = {{
    100, 101, 102, 103,
    104, 105, 106, 107,
    108, 109, 110, 112,
  }, {
    200, 201, 202, 203,
    204, 205, 206, 207,
    208, 209, 210, 212,
  }};